rt/rs/security/http-signature/src/main/java/org/apache/cxf/rs/security/httpsignature/provider/AlgorithmProvider.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.rs.security.httpsignature.provider;
+
+@FunctionalInterface
+public interface AlgorithmProvider {
+    /**
+     * @param keyId is used as lookup to find the correct configured algorithm name for this keyId
+     *              The keyId is sent in the message together with the signature
+     * @throws NullPointerException if it can't provide an algorithm based on keyId
```

Comment:
Might be better to use `IllegalAgumentException` if the provider can't provide an algorithm based on keyId